utils/portforward.go

```diff
@@ -0,0 +1,141 @@
+package utils
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"math/rand"
+	"net"
+	"net/http"
+	"net/url"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/kubearmor/kubearmor-client/k8s"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/tools/portforward"
+	"k8s.io/client-go/transport/spdy"
+)
+
+// Port forward details for a pod
+type PortForwardOpt struct {
+	LocalPort   int
+	RemotePort  int
+	MatchLabels map[string]string
+	Namespace   string
+	PodName     string
+}
+
+// Initialize port forwarding for kubearmor-relay pod
+func InitiatePortForward(c *k8s.Client, localPort int, remotePort int, matchLabels map[string]string) (PortForwardOpt, error) {
+	pf := PortForwardOpt{
+		LocalPort:   localPort,
+		RemotePort:  remotePort,
+		Namespace:   "",
+		MatchLabels: matchLabels,
+	}
+
+	// handle port forward
+	err := pf.HandlePortForward(c)
+	if err != nil {
+		return pf, err
+	}
+	return pf, nil
+}
+
+// handle port forward to allow grpc to connect at localhost:32767
+func (pf *PortForwardOpt) HandlePortForward(c *k8s.Client) error {
+	if err := pf.getPodName(c); err != nil {
+		return err
+	}
+
+	// local port
+	pf.LocalPort = pf.GetLocalPort()
+
+	err := k8sPortForward(c, *pf)
+	if err != nil {
+		fmt.Printf("could not do kubearmor portforward, Error=%s", err.Error())
+		return err
+	}
+	return nil
+
+}
+
+// k8s port forward
+func k8sPortForward(c *k8s.Client, pf PortForwardOpt) error {
+	roundTripper, upgrader, err := spdy.RoundTripperFor(c.Config)
+	if err != nil {
+		fmt.Printf("unable to spdy.RoundTripperFor error=%s", err.Error())
```

Comment:
It will display "spdy.RoundTripperFor" and then the error. Can we make it more clear?